Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs

```diff
@@ -29,10 +29,24 @@ private SubscriptingExpression(IExpression expression, IExpression index)
     public IExpression Lower(IDeclarationScope scope)
     {
         var expression = LowerExpression(_expression, scope);
-        var elementSize = GetElementSize(expression.GetExpressionType(scope));
+        var index = LowerExpression(_index, scope);
+        var expressionType = expression.GetExpressionType(scope);
+        var indexType = index.GetExpressionType(scope);
+
+        switch ((CheckIfTypeIsSubscriptable(expressionType), CheckIfTypeIsSubscriptable(indexType)))
```

Comment:
Because we are not exhaustive, maybe plain `if` will do the job? or maybe just introduction of variables for `CheckIfTypeIsSubscriptable(...)` calls. I think current logic a bit loaded.